modules/platform/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java

```diff
@@ -164,7 +165,7 @@ public void onOperationExit(Object output) {
 
             @Override
             public OperationException onError(OperationException error) {
-                setError(executionId, error.getMessage());
+                setError(executionId, ExceptionHelper.unwrapException(error).getMessage());
```

Comment:
I would have added a new method `setError(String, Throwable)` to handle the `unwrapException` logic (and deprecate the existing `setError` method probably as it won't be used anymore).